blockbase/inc/gutenberg-dependency-check.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+add_action( 'admin_notices', 'show_admin_messages' );
+
+function print_admin_message( $version ) {
+	echo '<div id="message" class="error"><p><strong>';
+	printf( __( 'The installed theme requires <a href="https://wordpress.org/plugins/gutenberg/">Gutenberg</a> version %s or higher.', 'livro' ), $version );
+	echo '</strong></p></div>';
+}
+
+function show_admin_messages() {
+
+	if ( is_child_theme() ) {
+		$metadata = file_get_contents( get_template_directory() . '/style.css' );
+		preg_match( '/(?<=Requires Gutenberg:).+/', $metadata, $match );
+	}
+
+	if ( sizeof( $match ) > 0 ) {
```

Comment:
I get an error that `$match` is undefined when I activate Blockbase: 

<img width="901" alt="Screen Shot 2022-03-24 at 2 31 02 PM" src="https://user-images.githubusercontent.com/5375500/159986229-4b2f9569-9c17-4d1d-9028-254a0dec9ccb.png">

Could we define `$metadata` before the child theme conditional, and overwrite it the conditional is met? Then right after this conditional but before the `sizeof` check, run `preg_match` to ensure $match is defined.

Reply:
Oh, odd that it didn't error for me...

I'm not sure how that `if( sizeof())` check snuck outside of the `is_child_theme()` check but I put that back in there where it belongs and you shouldn't error now.

We could indeed invert that logic as you suggest but that would be a larger change. It's otherwise 6 vs 1/2 dozen to me.